planner/core/common_plans.go

```diff
@@ -1151,6 +1172,137 @@ func (e *Explain) prepareDotInfo(p PhysicalPlan) {
 	e.Rows = append(e.Rows, []string{buffer.String()})
 }
 
+// JSONOperatorRow is the struct for a JSON formatted explain result.
+type JSONOperatorRow struct {
+	ID           string             `json:"id"`
+	EstRows      float64            `json:"estRows"`
+	Task         string             `json:"task"`
+	AccessObject string             `json:"accessObject"`
+	OperatorInfo string             `json:"operatorInfo"`
+	Children     []*JSONOperatorRow `json:"children"`
+}
+
+// prepareJSONOperatorInfo fills a JSONOperatorRow struct for the Plan p.
+func (e *Explain) prepareJSONOperatorInfo(currentNode *JSONOperatorRow, p Plan, taskType string, driverSide string) {
+	if p.ExplainID().String() == "_0" {
+		return
+	}
+	currentNode.ID = p.ExplainID().String() + driverSide
+	var (
+		tmpEstRows string
+		err        error
+	)
+	tmpEstRows, currentNode.AccessObject, currentNode.OperatorInfo = e.getOperatorInfo(p, currentNode.ID)
+	if currentNode.EstRows, err = strconv.ParseFloat(tmpEstRows, 64); err != nil {
+		currentNode.EstRows = 0
```

Comment:
Should we set `currentNode.Task` to `taskType`?

Reply:
I find the `task` field is empty in the result:
```
 {
  "id": "HashJoin_17",
  "estRows": 12487.5,
  "task": "",
  "accessObject": "",
  "operatorInfo": "inner join, equal:[eq(test.t2.t1_id, test.t1.id)]"
  ...
}
```

Reply:
Yes :-) Thanks for spotting.